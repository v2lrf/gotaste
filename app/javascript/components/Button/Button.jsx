import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import componentConfig from '../componentConfig'

function Button({ children, type, kind }) {
  const classNames = classnames(componentConfig.button.base, {
    [componentConfig.button.primary]: kind === 'primary',
    [componentConfig.button.secondary]: kind === 'secondary'
  })

  return (
    // eslint-disable-next-line react/button-has-type
    <button type={type} className={classNames}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  kind: PropTypes.oneOf(['primary', 'secondary']).isRequired
}

Button.defaultProps = {
  type: 'submit'
}

export default Button
