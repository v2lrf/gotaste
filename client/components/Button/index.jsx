import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

function Button({ children, type, kind }) {
  const classNames = classnames('px-6 h-10 rounded shadow tracking-wide', {
    'bg-red-darker text-white border border-red-darker hover:bg-red-darkest hover:border-red-darkest':
      kind === 'primary',
    'bg-white text-red-darker border border-red-darker hover:bg-red-darkest hover:text-white':
      kind === 'secondary'
  })

  return (
    // eslint-disable-next-line react/button-has-type
    <button type={type} className={classNames}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  kind: PropTypes.oneOf(['primary', 'secondary']).isRequired
}

Button.defaultProps = {
  type: 'submit'
}

export default Button
