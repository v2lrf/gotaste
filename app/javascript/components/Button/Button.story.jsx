import React from 'react'
import { storiesOf } from '@storybook/react'

import Button from './Button'

storiesOf('Button', module)
  .add('Primary', () => <Button kind="primary">Primary</Button>)
  .add('Secondary', () => <Button kind="secondary">Secondary</Button>)
