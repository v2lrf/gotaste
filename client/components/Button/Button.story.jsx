import React from 'react'
import { storiesOf } from '@storybook/react'

import Button from '.'

storiesOf('Button', module)
  .add('Primary', () => (
    <div className="p-16">
      <Button kind="primary">Primary</Button>
    </div>
  ))
  .add('Secondary', () => (
    <div className="p-16">
      <Button kind="secondary">Secondary</Button>
    </div>
  ))
