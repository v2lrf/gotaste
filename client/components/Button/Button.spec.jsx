import React from 'react'
import Button from '.'

describe('Button', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      kind: 'primary'
    }

    component = shallow(<Button {...props}>I am a button!</Button>)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
