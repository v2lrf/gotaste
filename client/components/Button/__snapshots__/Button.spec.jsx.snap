// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots should match the snapshot 1`] = `
<button
  className="px-6 h-10 rounded shadow tracking-wide bg-red-darker text-white border border-red-darker hover:bg-red-darkest hover:border-red-darkest"
  type="submit"
>
  I am a button!
</button>
`;
